### PR TITLE
Fix the command name, and version metadata, of the command installed by 'go install'.

### PR DESCRIPTION
**Pull request by AJChapman**
A bug I introduced the other day by changing to use `go install` instead of copying the built binary manually to $GOBIN/, which doesn't work if the user doesn't have GOBIN set.

The first issue was that the command installed by `make install` was `ivcap-cli`, but it should be `ivcap`. This is because `go install` doesn't allow you to rename the binary -- it will always be the name of the directory containing the main module of the command being installed, in this case `ivcap-cli`. The fix to this issue was to make an `ivcap` subdirectory and put the main package in there.

The next issue was that the version information inserted using the `-ldflags ...` command to
 `go build` was not making it into the installed binary. It turns out that `go install` doesn't
use the binary created by `go build`, it compiles it itself! The fix in this case was to pass the same `-ldflags ...` command to `go install`.

Note that the `install` make target now no longer depends on the `build` target, because it in fact does not use the binary that the `build` target creates.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,14 +15,15 @@ endif
...
 build: check
 	@echo "Building IVCAP-CLI..."
 	${GOPRIVATE_ENV_CMD} go mod tidy
-	go build -ldflags "-X main.version=${GIT_TAG} -X main.commit=${GIT_COMMIT} -X main.date=${BUILD_DATE}" -o ivcap${EXTENSION}
+	go -C ivcap build -ldflags ${LD_FLAGS}
 
-install: addlicense build
-	$(shell go install)
+install: addlicense check
+	go -C ivcap install -ldflags ${LD_FLAGS}
```
Comment (johnzhanghua) [inline Makefile:26]: Does that work? don't think there is `ivcap` dir.

Comment (AJChapman) [reply Makefile:27]: Yes, it works. This patch also moves cli.go to ivcap/cli.go, which is required by the go tooling.

Comment (johnzhanghua) [reply Makefile:27]: Ahh, yeah, there is a folder change.  Sorry ignored that. 

I think its better to be put under cmd/ivcap .

**Review by johnzhanghua — APPROVED** [review]
LGTM.

**Pull request merged**